doc/holochain_101/src/planning_a_dapp.md

```diff
@@ -1 +1,68 @@
 # Planning a dApp
+
+## Introduction
+Generally speaking, you need to know the following in order to build a holochain dApp:
+
+   how to install holochain
+   
+   how to use the command line tools
+   
+   how to configure your application with a "DNA" file
+   
+   how to write your application code in JavaScript or Lisp
+   
+<b>how to think through building a distributed application</b> 
+
+   how to build a user interface for your app
+   
+   how to test your application code
+
+This article will help you plan a dApp by providing practical considerations about the specifics of distributed applications in general, and holochain dApps in particular. It has been remarked that holochain dApps require us to make a mental shift, first from applications whose data is centrally organized, and also from blockchain-based, data-centric dApps.
+
+Here we will provide a basic overview of concepts from cryptography that are central to holochains. 
+Then, we will consider the consequences of holochain's cryptographic architecture for data permissioning, access, and security. 
+
+## What is a dApp? 
+A dApp is a distributed application. This means that the data associated with the application is stored by each user rather than in a central database.  
+
+## Basic expectations for dApps
+Because app data storage is distributed amongst user-participants, one must expect that data encryption and permissions are important for protecting privacy in accordance with the jurisdictions in which the app is operating.
+
+Remember that, as user-participants leave the application, they take their data with them. They also retain copies of other data that they held to support the DHT.  
+
+One must also re-think the dApp's business model such that it does not rely on a central authority's ability to whitelist access to a given resource. 
+
+## Cryptography in holochain dApps
+Distributed systems rely more heavily on cryptography than centralized systems. 
+
+### Hashes
+
+### Signatures 
+
+### Encryption
+
+## Entry types 
+You can define particular entry types as public, private, or encrypted. Encryption is an in-built option for holochain apps, meaning that you can automatically encrypt entry types by default.
+This is because holochain's use of DHTs means that simply running the application allows user-participants access to all entries on the network.
+
+### Public entries
+Public entries are published to the app's DHT with an appropriate redundancy factor. 
+
+### Encrypted entries
+Like public entries, encrypted entries are published to the app's DHT with an appropriate redundancy factor.
+
+### Private entries
+Unlike public and encrypted entries, private entries are only on the source chain (device) of the publisher and are not shared out to the DHT. 
+Of course, one could always encrypt a private entry, too. 
+Consider the use cases of a private blockchain...
```

Comment:
Hm, not sure how the concept of private blockchains fit in here..

Reply:
Yup. Misplaced future intent to explore those use cases -- would work better, if at all, in discussion of membranes, as you mention below !